### DISCUSSION TITLE
Intent and use cases behind EntityManagerDecorator

### DISCUSSION THREAD

**Discussion by dennis-fedco** [doctrine/doctrine2#8839] [Support Questions] [answered]
I am having a hard time finding documentation on `EntityManagerDecorator`.

Can you please explain what `EntityManagerDecorator` is used for?
I am looking for a good typical expected use case example of using the `EntityManagerDecorator`.
And in particular if it has any context of being used with classes that extend `EntityRepository`.

I also want to add something which I think is an incorrect usage of the decorator to give you more context behind my question on understanding how to use the decorator correctly:    

```php
class DoctrineProductRepository extends EntityManagerDecorator
{
    function getProduct(int $id): Product
    {
        return $this->getRepository(Product::class)->find($id);
    }
}

// to call
$repository = new DoctrineProductRepository($this->getEntityManager());
$repository->getProduct(2); //Product entity

**Comment by beberlei** [comment, marked as answer]
You don't need the decorator, see how to configure custom repositories in mapping. You need a ProductRepository extends EntityRepository
